Readme.md

```diff
@@ -1,176 +1,341 @@
+
+This allows a user to stop and/or delete the index/logging of specific domains or pages they visit.
+
+#### **[src/common-ui/components](src/common-ui/components)**: ui elements
+
+This contains common user interface elements such as the loading indicator. 
+
+#### **[src/dev/](src/dev/)**: development tools
+
+Tools to help during development. They are not used in production builds.
+
+#### **[src/imports/](src/imports/)**: browser history import
+
+This allows users to import their whole browser history however, due to slow speeds and multiple setbacks it has been deprecated and may only be kept on as a dev tool to import test docs. 
+
+Instead of using a browser this module is set to move to a native Desktop Apllication see [roadmap][roadmap] for more details
+
+#### **[src/options](src/options/)**: the settings page
+
+This shows the settings page of the extension which includes (for the time being)
+ - Blacklist
+ - Acknowledgements 
+ - Privacy
+ - Help Me Please
+
+#### **[src/overview/](src/overview/)**: overview
+
+The `overview` is a user interface that opens in its own tab. It shows a 'facebook newsfeed' like page that lists random pages you have visited. It also provides tools for more advanced search options and a complete search overview with screenshots. It is built with [React](https://facebook.github.io/react/) and [Redux](http://redux.js.org/).
+
+See [The Docs](src/overview/Readme.md) for more details.
+
+#### **[src/page-analysis](src/page-analysis/)**: page analysis
+
+This extracts and stores information about the page in a given tab, such as:
+- The plain text of the page, mainly for the full-text
+search index.
+- Metadata, such as its author, publication date, etc...
+- A screenshot for visual recognition.
+
+See [The Docs](src/page-analysis/Readme.md) for more details.
+
+#### **[src/page-storage](src/page-storage/)**: document search
+
+This runs through each page to store and makes sure there isn't a duplicate and prepares it for use in our data model 
+
+See [The Docs](src/page-storage/Readme.md) for more details.
+
+#### **[src/popup](src/popup/)**: extension popup
+
+The `popup` is a mini UI that pops up when you click on the worldbrain. It contains
+- search
+- pause
+- settings
+- feedback
+
+See [The Docs](src/popup/Readme.md) for more details.
+
+#### **[src/search](src/search/)**: document search
+
+This provides a **Full-Text-Search** through all the web pages a user visits. 
+
+See [The Docs](src/search/Readme.md) for more details.
+
+#### **[src/util/](src/util)**: utilities
+
+Contains small generic things, stuff that is not project-specific. Things that
+could perhaps be packaged and published as a NPM modules some day.
+See [The Docs](src/util/Readme.md) for more details.
+
+#### **[src/background.js](src/background.js)**: WebExtension background script
+
+This provides a wrapper for all the process run in the background which sit in an empty tab listening for events.

```

Comment:
Now that the bookmarks stuff has been refactored into its own module, `src/background.js` is simply just an entrypoint for the main WebExt background script that pull in all the other background parts from the codebase (and set up some event listeners, as noted). `src/content_script.js` is essentially the same thing but for the main WebExt content script. So I think no need to list those functions. 

But could add another section for `src/bookmarks` module. Maybe:

<pre lang="no-highlight"><code>
#### **[src/bookmarks](src/bookmarks)**: Bookmarks-related logic
 
Contains all the logic related to handling our data structure for bookmarks, along with background event listeners for handling browser bookmark creation.
</code></pre>